File: app/routes/health_routes.py
import os
import sys
import time
from datetime import datetime
from fastapi import APIRouter, HTTPException, status
from pydantic import BaseModel

from app.services.health_service import check_database_connections

router = APIRouter()

# Pydantic models
class HealthResponse(BaseModel):
    status: str
    timestamp: str
    service: str
    version: str
    databases: dict
    uptime: float

class DetailedHealthResponse(BaseModel):
    status: str
    timestamp: str
    service: str
    version: str
    environment: str
    databases: dict
    system: dict

@router.get("/", response_model=HealthResponse)
async def health_check():
    """
    Basic health check endpoint
    """
    try:
        health_status = await check_database_connections()
        return HealthResponse(
            status="healthy",
            timestamp=datetime.now().isoformat(),
            service="CONTRACTEXTRACT AI Agent",
            version="1.0.0",
            databases=health_status,
            uptime=time.time()  # Simplified uptime calculation
        )
    except Exception as error:
        print(f"Health check failed: {error}")
        raise HTTPException(
            status_code=status.HTTP_503_SERVICE_UNAVAILABLE,
            detail={
                "status": "unhealthy",
                "timestamp": datetime.now().isoformat(),
                "service": "CONTRACTEXTRACT AI Agent",
                "error": str(error),
                "uptime": time.time()  # Simplified uptime calculation
            }
        )

@router.get("/detailed", response_model=DetailedHealthResponse)
async def detailed_health_check():
    """
    Detailed health check with system information
    """
    try:
        health_status = await check_database_connections()
        # Use simple placeholders for memory and cpu
        memory_usage = {
            "total": "N/A",
            "available": "N/A",
            "percent": "N/A"
        }
        cpu_info = {
            "count": "N/A",
            "percent": "N/A"
        }
        return DetailedHealthResponse(
            status="healthy",
            timestamp=datetime.now().isoformat(),
            service="CONTRACTEXTRACT AI Agent",
            version="1.0.0",
            environment=os.getenv("NODE_ENV", "development"),
            databases=health_status,
            system={
                "uptime": time.time(),  # Simplified uptime calculation
                "memory": memory_usage,
                "cpu": cpu_info,
                "platform": os.name,
                "python_version": f"{sys.version_info.major}.{sys.version_info.minor}.{sys.version_info.micro}"
            }
        )
    except Exception as error:
        print(f"Detailed health check failed: {error}")
        raise HTTPException(
            status_code=status.HTTP_503_SERVICE_UNAVAILABLE,
            detail={
                "status": "unhealthy",
                "timestamp": datetime.now().isoformat(),
                "service": "CONTRACTEXTRACT AI Agent",
                "error": str(error),
                "uptime": time.time()  # Simplified uptime calculation
            }
        ) 

File: src/routes/healthRoutes.js
const express = require('express');
const router = express.Router();
const { checkDatabaseConnections } = require('../services/healthService');

// Health check endpoint
router.get('/', async (req, res) => {
  try {
    const healthStatus = await checkDatabaseConnections();
    
    res.json({
      status: 'healthy',
      timestamp: new Date().toISOString(),
      service: 'CONTRACTEXTRACT AI Agent',
      version: '1.0.0',
      databases: healthStatus,
      uptime: process.uptime()
    });
  } catch (error) {
    console.error('Health check failed:', error);
    res.status(503).json({
      status: 'unhealthy',
      timestamp: new Date().toISOString(),
      service: 'CONTRACTEXTRACT AI Agent',
      error: error.message,
      uptime: process.uptime()
    });
  }
});

// Detailed health check
router.get('/detailed', async (req, res) => {
  try {
    const healthStatus = await checkDatabaseConnections();
    const memoryUsage = process.memoryUsage();
    
    res.json({
      status: 'healthy',
      timestamp: new Date().toISOString(),
      service: 'CONTRACTEXTRACT AI Agent',
      version: '1.0.0',
      environment: process.env.NODE_ENV || 'development',
      databases: healthStatus,
      system: {
        uptime: process.uptime(),
        memory: {
          rss: `${Math.round(memoryUsage.rss / 1024 / 1024)} MB`,
          heapTotal: `${Math.round(memoryUsage.heapTotal / 1024 / 1024)} MB`,
          heapUsed: `${Math.round(memoryUsage.heapUsed / 1024 / 1024)} MB`,
          external: `${Math.round(memoryUsage.external / 1024 / 1024)} MB`
        },
        nodeVersion: process.version,
        platform: process.platform
      }
    });
  } catch (error) {
    console.error('Detailed health check failed:', error);
    res.status(503).json({
      status: 'unhealthy',
      timestamp: new Date().toISOString(),
      service: 'CONTRACTEXTRACT AI Agent',
      error: error.message,
      uptime: process.uptime()
    });
  }
});

module.exports = router; 